src/Bicep.Core/Semantics/Namespaces/AzNamespaceType.cs

```diff
@@ -330,52 +349,65 @@ private static IEnumerable<FunctionOverload> GetAzOverloads(ResourceScope resour
                 .WithReturnType(LanguageConstants.String)
+                .WithGenericDescription(subscriptionResourceIdDescription)
                 .WithDescription(subscriptionResourceIdDescription)
                 .WithRequiredParameter("resourceType", LanguageConstants.String, "Type of resource including resource provider namespace")
                 .WithVariableParameter("resourceName", LanguageConstants.String, minimumCount: 1, "The resource name segment")
                 .Build();
 
             yield return new FunctionOverloadBuilder("subscriptionResourceId")
                 .WithReturnType(LanguageConstants.String)
+                .WithGenericDescription(subscriptionResourceIdDescription)
                 .WithDescription(subscriptionResourceIdDescription)
                 .WithRequiredParameter("subscriptionId", LanguageConstants.String, "The subscription ID")
                 .WithRequiredParameter("resourceType", LanguageConstants.String, "Type of resource including resource provider namespace")
                 .WithVariableParameter("resourceName", LanguageConstants.String, minimumCount: 1, "The resource name segment")
                 .Build();
 
+            const string tenantResourceIdDescription = "Returns the unique identifier for a resource deployed at the tenant level.";
             yield return new FunctionOverloadBuilder("tenantResourceId")
                 .WithReturnType(LanguageConstants.String)
-                .WithDescription("Returns the unique identifier for a resource deployed at the tenant level.")
+                .WithGenericDescription(tenantResourceIdDescription)
+                .WithDescription(tenantResourceIdDescription)
                 .WithRequiredParameter("resourceType", LanguageConstants.String, "Type of resource including resource provider namespace")
                 .WithVariableParameter("resourceName", LanguageConstants.String, minimumCount: 1, "The resource name segment")
                 .Build();
 
+            const string extensionResourceIdDescription = "Returns the resource ID for an [extension](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/extension-resource-types) resource, which is a resource type that is applied to another resource to add to its capabilities.";
             yield return new FunctionOverloadBuilder("extensionResourceId")
                 .WithReturnType(LanguageConstants.String)
-                .WithDescription("Returns the resource ID for an [extension](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/extension-resource-types) resource, which is a resource type that is applied to another resource to add to its capabilities.")
+                // .WithGenericDescription("Returns the resource ID for an extension resource, which is a resource type that is applied to another resource to add to its capabilities.")
+                .WithGenericDescription(extensionResourceIdDescription)
```

Comment:
Could we also set `Description` in `WithGenericDescription`? That will remove the need to call `WithDescription` with the same const string.

Reply:
+1